modules/c++/nitf/unittests/test_image_segment_blank_nm_compression.cpp

```diff
@@ -0,0 +1,296 @@
+   imgSubHdr.insertImageComment("ABCDEFGHIJKLMNOPQRSTUVWXYZ", 0);
+   std::vector<nitf::BandInfo> bands;
+   nitf::BandInfo bandI;
+   bandI.getRepresentation().set("M ");
+   bandI.getSubcategory().set("      ");
+   bandI.getImageFilterCondition().set("N");
+   bandI.getImageFilterCode().set("   ");
+   bandI.getNumLUTs().set(0);
+   bands.push_back(bandI);
+   const std::string pixelValueType = "INT";
+   const nitf::Int64 numBitsPerPixel = 8;
+   const nitf::Int64 actualBitsPerPixel = 8;
+   const std::string pixelJustification = "R";
+   const std::string imageRepresentation = "MONO";
+   const std::string imageCategory = "SOS";
+
+   imgSubHdr.setPixelInformation(
+      pixelValueType,
+      numBitsPerPixel,
+      actualBitsPerPixel,
+      pixelJustification,
+      imageRepresentation,
+      imageCategory,
+      bands);
+   imgSubHdr.getImageSyncCode().set("0");
+   imgSubHdr.getImageMode().set("B");
+   nitf::Int64 blocksPerRow = (segmentDims.col + (BLOCK_LENGTH - 1)) / BLOCK_LENGTH;
+   nitf::Int64 blocksPerCol = (segmentDims.row + (BLOCK_LENGTH - 1)) / BLOCK_LENGTH;
+
+   imgSubHdr.getNumBlocksPerRow().set(blocksPerRow);
+   imgSubHdr.getNumPixelsPerHorizBlock().set(BLOCK_LENGTH);
+   imgSubHdr.getNumBlocksPerCol().set(blocksPerCol);
+   imgSubHdr.getNumPixelsPerVertBlock().set(BLOCK_LENGTH);
+
+   imgSubHdr.getImageDisplayLevel().set((nitf::Int64)imageIndex+1);
+   imgSubHdr.getImageAttachmentLevel().set((nitf::Int64)imageIndex);
+
+   imgSubHdr.getImageLocation().set(generateILOC(segmentOffset));
+
+   return imgSubHdr;
+}
+
+nitf::Int64 getNumberBlocksPresent(const nitf_Uint64 *mask,
+                                   const nitf::Int64 numRows,
+                                   const nitf::Int64 numCols)
+{
+   nitf::Int64 numBlocksPresent = 0;
+   for (nitf::Int64 row = 0, idx = 0; row < numRows; ++row)
+   {
+      for (nitf::Int64 col = 0; col < numCols; ++col, ++idx)
+      {
+         if (mask[idx] != 0xFFFFFFFF)
+         {
+            ++numBlocksPresent;
+         }
+      }
+   }
+
+   return numBlocksPresent;
+}
+
+void createSingleBandBuffer(std::vector<nitf::Uint8> &buffer,
+                            const nitf::ImageSegmentComputer &segmentComputer,
+                            const types::RowCol<nitf::Int64> &fullDims,
+                            const nitf::Int64 segmentIdxToMakeEmpty)
+{
+   const nitf::Int64 bytes = fullDims.area();
+   const std::vector<nitf::ImageSegmentComputer::Segment> &segments = segmentComputer.getSegments();
+   /* All segments should be the same size in this test so this is safe */
+   const nitf::Int64 segmentSizeInBytes = segments[segmentIdxToMakeEmpty].numRows * fullDims.col;
+
+   buffer.resize(bytes);
+   memset(&buffer.front(), '0xff', bytes);
+   memset(&buffer.front() + (segmentSizeInBytes * segmentIdxToMakeEmpty),
+          '\0', segmentSizeInBytes);
+}
+
+void createBuffers(std::vector<std::vector<nitf::Uint8> > &buffers,
```

Comment:
And then make sure the references get pushed to the left too, whenever they pop up. (`std::vector<nitf::Uint8> >& buffers`)